guides/index_lifecycle.md

```diff
@@ -0,0 +1,157 @@
+# Index Lifecycle
+
+This guide covers OpenSearch Python Client API actions for Index Lifecycle. You'll learn how to create, read, update, and delete indices in your OpenSearch cluster. We will also leverage index templates to create default settings and mappings for indices of certain patterns.
+
+## Setup
+
+In this guide, we will need an OpenSearch cluster with more than one node. Let's use the sample [docker-compose.yml](https://opensearch.org/samples/docker-compose.yml) to start a cluster with two nodes. The cluster's API will be available at `localhost:9200` with basic authentication enabled with default username and password of `admin:admin`.
+
+To start the cluster, run the following command:
+
+```bash
+cd /path/to/docker-compose.yml
+docker-compose up -d
+```
+
+Let's create a client instance to access this cluster:
+
+```python
+from opensearchpy import OpenSearch
+
+client = OpenSearch(
+    hosts=['https://admin:admin@localhost:9200'],
+    use_ssl=True,
+    verify_certs=False
+)
+
+print(client.info())  # Check server info and make sure the client is connected
+```
+
+## Index API Actions
+
+### Create a new index
+
+You can quickly create an index with default settings and mappings by using the `indices.create` API action. The following example creates an index named `paintings` with default settings and mappings:
+
+```python
+client.indices.create(index='paintings')
+```
+
+To specify settings and mappings, you can pass them as the `body` of the request. The following example creates an index named `movies` with custom settings and mappings:
+
+```python
+client.indices.create(
+    index='movies',
+    body={
+        'settings': {
+            'index': {
+                'number_of_shards': 2,
+                'number_of_replicas': 1
+            }
+        },
+        'mappings': {
+            'properties': {
+                'title': {'type': 'text'},
+                'year': {'type': 'integer'}
+            }
+        }
+    }
+)
+```
+
+When you create a new document for an index, OpenSearch will automatically create the index if it doesn't exist:
+
+```python
+print(client.indices.exists(index='burner'))  # => False
+client.index(id='1', index='burner', body={'lorem': 'ipsum'})
+print(client.indices.exists(index='burner'))  # => True
+```
+
+### Update an Index
+
+You can update an index's settings and mappings by using the `indices.put_settings` and `indices.put_mapping` API actions.
+
+The following example updates the `movies` index's number of replicas to `0`:
+
+```python
+client.indices.put_settings(
+      index='movies',
+      body={
+            'index': {
```

Comment:
Some of the indentations in this file look a little off. Can you correct them?

Reply:
Done! Please check :)